modules/foreman_global_parameter.py

```diff
@@ -174,38 +157,36 @@ def main():
             with: pip install nailgun"""
         )
 
-    server_url = module.params['server_url']
-    username = module.params['username']
-    password = module.params['password']
-    verify_ssl = module.params['verify_ssl']
-    name = module.params['name']
-    value = module.params['value']
-    state = module.params['state']
-    auth = (username, password)
-    changed = False
+    global_parameter_dict = dict(
+        [(k, v) for (k, v) in module.params.iteritems() if v is not None])
+
+    server_url = global_parameter_dict.pop('server_url')
+    username = global_parameter_dict.pop('username')
+    password = global_parameter_dict.pop('password')
+    verify_ssl = global_parameter_dict.pop('verify_ssl')
+    state = global_parameter_dict.pop('state')
+
     try:
-        createServer(server_url, auth, verify_ssl)
-        param = findEntity(CommonParameter, name=name)
+        create_server(server_url, (username, password), verify_ssl)
     except Exception as e:
         module.fail_json(msg="Failed to connect to Foreman server: %s " % e)
+
+    ping_server(module)
     try:
-        if state == 'present':
-            if len(param) == 1:
-                if param[0].value != value:
-                    param[0].value = value
-                    param[0].update()
-                    changed = True
-            else:
-                createCommonParameter(name, value)
-                changed = True
-        elif state == 'absent':
-            if len(param) == 1:
-                param[0].delete()
-                changed = True
+        entities = find_entities(CommonParameter, name=global_parameter_dict['name'])
+        if len(entities) > 0:
+            entity = entities[0]
+        else:
+            entity = None
     except Exception as e:
-        module.fail_json(msg=e)
+        module.fail_json(msg='Failed to find entity: %s ' % e)
+
+    global_parameter_dict = sanitize_global_parameter_dict(global_parameter_dict)
+
+    changed = naildown_entity_state(CommonParameter, global_parameter_dict, entity, state, module)
```

Comment:
This changes the behavior of `state: present` as discussed in #30. But the examples still use `present` for an update ;)

Also, `latest` is missing from the doc in the header.